llvm/lib/Target/PowerPC/PPCAsmPrinter.cpp

```diff
@@ -1523,19 +1558,50 @@ void PPCAsmPrinter::emitInstruction(const MachineInstr *MI) {
     EmitToStreamer(*OutStreamer, MCInstBuilder(PPC::EnforceIEIO));
     return;
   }
+  case PPC::LBZ:
```

Comment:
just curiously: 
tls  addi, or load/stores with thread-pointer are only in 64 bits, doese  PPC:LBZ is 32bits? 

same as :LHA  etc..

Reply:
Yeah, I found that these versions (the 32-bit, non-`8` versions of the instructions) can still appear.

They can get selected if one of their the preceding operands are 32-bit versions instructions. For example, below is a segment of the SDAG where we can select both `STB` and `STB8`. The `STB` gets selected since one of the operands are an `ADDI`.
```
. . . 
          t85: i64 = LI8 TargetConstant:i64<10>
        t62: ch = STB8<Mem:(store (s8) into %ir.0, !tbaa !3)> t85, TargetGlobalTLSAddress:i64<ptr @IThreadLocalVarInit> 0 [TF=7], Register:i64 $x13, t0
          t84: i64 = LI8 TargetConstant:i64<-46>
        t74: ch = STB8<Mem:(store (s8) into %ir.arrayidx1, !tbaa !3)> t84, TargetGlobalTLSAddress:i64<ptr @IThreadLocalVarInit> + 10 [TF=7], Register:i64 $x13, t0
          t60: i32 = ADDI t55, TargetConstant:i32<2>
        t61: ch = STB<Mem:(store (s8) into %ir.arrayidx3, !tbaa !3)> t60, TargetGlobalTLSAddress:i64<ptr @IThreadLocalVarInit> + 20 [TF=7], Register:i64 $x13, t0
. . .
```

Reply:
My question is "the thread pointer for tls r13 is only for 64bit , does the instruction will be selected PPC::LBZ when the compiler target is 64bit?"

Reply:
I could be mistaken, but to my knowledge, similar to the previous comment I added, I believe `PPC::LBZ` and these other 32-bit version of instructions can be selected on a 64-bit target.

On a 32-bit target for `PPC::LBZ`, it will go into this case but it should end up in this case and break out to process the instruction normally.
```
    // Now process the instruction normally.
    break;
```
The part you mentioned:
```
if (Flag == PPCII::MO_TPREL_FLAG ||
    Flag == PPCII::MO_GOT_TPREL_PCREL_FLAG ||
    Flag == PPCII::MO_TPREL_PCREL_FLAG) {
  assert(HasAIXSmallLocalExecTLS &&
               "addi, or load/stores with thread-pointer only expected with "
               "local-exec small TLS");
```
Should happen for 64-bit targets, and that assert should be true for 64-bit only because we have restricted the aix-small-local-exec-tls feature to 64-bit only in https://github.com/llvm/llvm-project/commit/b1922e55ab3b35dff99238fd0b74be00df0472e7 and in https://github.com/llvm/llvm-project/commit/8bdbee8aaa76960c54cd0071455b92e9d60d8966.

Reply:
thanks for confirm the PPC::LBZ